scripts/js/lib/api/processHtml.test.ts

```diff
@@ -313,6 +314,29 @@ test("addLanguageClassToCodeBlocks()", () => {
     </div>`);
 });
 
+test("updateRedirectedExternalLinks()", () => {
+  const redirects = {
+    "https://ibm.com/old": "https://ibm.com/new",
+    "https://basename.example-website.com/old#anchor?and=query":
+      "https://newbasename.example.com/new#newanchor?query=new",
+  };
+  const doc = CheerioDoc.load(
+    `<a href="https://ibm.com/old">https://ibm.com/old</a>
+    <a href="https://ibm.com/old">https://ibm.com/old (v3)</a>
+    <a href="https://ibm.com/old-but-not-redirected">https://ibm.com/old-but-not-redirected (v3)</a>
+    <a href="https://basename.example-website.com/old#anchor?and=query"></a>
+    <p>https://ibm.com/old</p>`,
+  );
+  updateRedirectedExternalLinks(doc.$, doc.$main, redirects);
+  doc.expectHtml(
+    `<a href="https://ibm.com/new">https://ibm.com/new</a>
+    <a href="https://ibm.com/new">https://ibm.com/new (v3)</a>
+    <a href="https://ibm.com/old-but-not-redirected">https://ibm.com/old-but-not-redirected (v3)</a>
```

Comment:
Given how your regex has `(?=$|[\\s<'"])`, it would maybe help to also test 

a) text is the exact non-replaced URL
b) text is the exact non-replaced URL with a query parameter

Reply:
Is this what you mean? da6b62efa64d4951027cd35a0241da54f6d15774